src/openrct2/Editor.cpp

```diff
@@ -562,7 +562,7 @@ namespace Editor
     {
         if (index != OBJECT_ENTRY_INDEX_NULL)
         {
-            assert(static_cast<int32_t>(objectType) < object_entry_group_counts[EnumValue(ObjectType::Paths)]);
+            assert(index < getObjectEntryGroupCount(ObjectType::Paths));
```

Comment:
This appears to have been a bug. Comparing an object type to the max group count does not make sense; I'm assuming index was meant.